### ISSUE TITLE
Property Collection Associations links to 404

### ISSUE THREAD

**Issue by lmtoo** [paulcwarren/spring-content#90]
I defined a DocumentGroup to Document with OneToMany relation, but DocumentGroup's link http://localhost:8080/api/v1/content/2/entries/882ab7b0-545f-4a2a-b727-43d992978d2d always be 404.

```
{
    "_embedded": {
        "documentGroups": [
            {
                "name": "groupName",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/api/v1/documentGroup/2"
                    },
                    "documentGroup": {
                        "href": "http://localhost:8080/api/v1/documentGroup/2"
                    },
                    "entries": [
                        {
                            "href": "http://localhost:8080/api/v1/content/2/entries/882ab7b0-545f-4a2a-b727-43d992978d2d"
                        },
                        {
                            "href": "http://localhost:8080/api/v1/documentGroup/2/entries"
                        }
                    ]
                }
            }
        ]
    },
    "_links": {
        "self": {
            "href": "http://localhost:8080/api/v1/documentGroup/"
        },
        "profile": {
            "href": "http://localhost:8080/api/v1/profile/documentGroup"
        }
    }
}
```

reproduce address https://github.com/lmtoo/spring-content-demos/tree/issues-90

**Comment by paulcwarren** [comment]
This link is a "content property" link.  These links were designed around `@Embeddable` JPA Entities and Mongodb `@Documents`.  They don't work for all use cases.  This being one.  

So I think the bug here is that the `ContentLinksProcessor` is emitting these links in a non-`@Embeddable` scenario.  The fix is to remove this link.   

If you are using these links to navigate your client about (i.e. you have a hypermedia client) then you would get to the content by following the second "entries" link; i.e. the `http://localhost:8080/api/v1/documentGroup/2/entries` one and then following it's "content" link.

**Comment by paulcwarren** [comment]
Fixed with [this commit](ce8decc276dda1395fa180a09dbe0d2597b97302).

Decided that content property links were fundamentally flawed so they will no longer be generated.  To get content, hypermedia clients should follow links to the actual entities and use their content links instead.
